lib/dma.c

```diff
@@ -608,6 +804,11 @@ dma_controller_dirty_page_get(dma_controller_t *dma, vfu_dma_addr_t addr,
         return ERROR_INT(EINVAL);
     }
 
+    if (region->dirty_bitmap == NULL) {
```

Comment:
Before, logging was either enabled for all regions (which we can check by `dma->dirty_pgsize != 0`) or none, so if we got past the previous check of `if (pgsize != dma->dirty_pgsize) { ...` we knew that the dirty bitmap would not be null. However, in this commit you were able to start logging only on specific regions, so we need the additional check.

Since we're getting rid of this optimisation for now and going back to the "all-or-nothing" approach to dirty page logging, I'll get rid of this too.